### DISCUSSION TITLE
Json::parse from incorrect string causes segfault

### DISCUSSION THREAD

**Discussion by devmsb** [wfrest/wfrest#159] [Q&A] [answered]
Hi,

Is Json have any method to check data format for valid JSON data before parsing?

Calling 
`Json Data = Json::parse("invalid text")`
causes segfault on any use of Data variable.

**Comment by chanchann** [comment, marked as answer]
<img width="693" alt="image" src="https://user-images.githubusercontent.com/27816166/235443357-f66a822f-ee4f-42ec-871e-5958d3580172.png">
<img width="896" alt="image" src="https://user-images.githubusercontent.com/27816166/235443413-15d6f308-efb8-4416-bdec-2c7e42ed43c0.png">

**Reply by devmsb** [reply]
Thank you! I used method from examples `req->json<Json>()` and it causes segfault. 
`Json::parse(req->body())` works perfectly!
